### PR TITLE
2) test cases for mock time entry controller API

### PR DESCRIPTION
**Pull request by farhaninfyom**

### CODE REVIEW

File: tests/Controllers/TimeEntryControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    {
+        $this->mockRepository();
+
+        /** @var TimeEntry $timerEntry */
+        $timeEntry = factory(TimeEntry::class)->create();
+
+        $this->timeEntryRepository->shouldReceive('getTimeEntryDetail')
+            ->once()
+            ->with($timeEntry->id)
+            ->andReturn([]);
```
Comment (vishalinfyom) [inline tests/Controllers/TimeEntryControllerTest.php]: remove this from everywhere as we are not using it

File: tests/Controllers/TimeEntryControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+        $response->assertStatus(404);
+        $response->assertJson([
+            'success' => false,
+            'message' => 'TimeEntry not found.',
+        ]);
+    }
+
+    /** @test */
+    public function test_can_get_user_last_task()
```
Comment (vishalinfyom) [inline tests/Controllers/TimeEntryControllerTest.php]: `test_can_get_last_task_details_of_logged_in_user`

File: tests/Controllers/TimeEntryControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    {
+        $this->mockRepository();
+
+        $this->timeEntryRepository->shouldReceive('myLastTask')
+            ->once()
+            ->with();
```
Comment (vishalinfyom) [inline tests/Controllers/TimeEntryControllerTest.php]: with is only for this function in which we need to pass param, and this function don't need any param so remove it.

File: tests/Controllers/TimeEntryControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        /** @var TimeEntry $timerEntry */
+        $timeEntry = factory(TimeEntry::class)->create();
+
+        $this->timeEntryRepository->shouldReceive('getTimeEntryDetail')
+            ->once()
+            ->with($timeEntry->id);
+
+        $response = $this->getJson("time-entries/$timeEntry->id/edit");
+
+        $this->assertSuccessMessageResponse($response, 'Time Entry retrieved successfully.');
```
Comment (mitulgolakiya) [inline tests/Controllers/TimeEntryControllerTest.php]: we need to assert time entry details here and all other places where a controller is returning a data in the response.

**Pull request merged**